integration-tests/tests/singlestore/singlestore-common.ts

```diff
@@ -0,0 +1,3440 @@
+	dateAsString: date('date_as_string', { mode: 'string' }),
+	time: time('time', { fsp: 1 }),
+	datetime: datetime('datetime', { fsp: 6 }),
+	datetimeAsString: datetime('datetime_as_string', { fsp: 6, mode: 'string' }),
+	timestamp: timestamp('timestamp', { fsp: 6 }),
+	timestampAsString: timestamp('timestamp_as_string', { fsp: 6, mode: 'string' }),
+	year: year('year'),
+});
+
+const coursesTable = singlestoreTable('courses', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	categoryId: int('category_id'),
+});
+
+const courseCategoriesTable = singlestoreTable('course_categories', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+});
+
+const orders = singlestoreTable('orders', {
+	id: serial('id').primaryKey(),
+	region: text('region').notNull(),
+	product: text('product').notNull().$default(() => 'random_string'),
+	amount: int('amount').notNull(),
+	quantity: int('quantity').notNull(),
+});
+
+const usersMigratorTable = singlestoreTable('users12', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	email: text('email').notNull(),
+}, (table) => {
+	return {
+		name: uniqueIndex('').on(table.name).using('btree'),
+	};
+});
+
+// To test aggregate functions
+const aggregateTable = singlestoreTable('aggregate_table', {
+	id: serial('id').notNull(),
+	name: text('name').notNull(),
+	a: int('a'),
+	b: int('b'),
+	c: int('c'),
+	nullOnly: int('null_only'),
+});
+
+// To test another schema and multischema
+const mySchema = singlestoreSchema(`mySchema`);
+
+const usersMySchemaTable = mySchema.table('userstest', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	verified: boolean('verified').notNull().default(false),
+	jsonb: json('jsonb').$type<string[]>(),
+	createdAt: timestamp('created_at', { fsp: 6 }).notNull().defaultNow(),
+});
+
+const users2MySchemaTable = mySchema.table('users2', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+	cityId: int('city_id'),
+});
+
+const citiesMySchemaTable = mySchema.table('cities', {
+	id: serial('id').primaryKey(),
+	name: text('name').notNull(),
+});
+
+let singlestoreContainer: Docker.Container;
+export async function createDockerDB(): Promise<{ connectionString: string; container: Docker.Container }> {
+	const docker = new Docker();
+	const port = await getPort({ port: 3306 });
+	const image = 'ghcr.io/singlestore-labs/singlestoredb-dev:latest';
+
+	const pullStream = await docker.pull(image);
+	await new Promise((resolve, reject) =>
+		docker.modem.followProgress(pullStream, (err) => (err ? reject(err) : resolve(err)))
+	);
+
+	singlestoreContainer = await docker.createContainer({
+		Image: image,
+		Env: ['ROOT_PASSWORD=singlestore'],
+		name: `drizzle-integration-tests-${uuid()}`,
+		HostConfig: {
+			AutoRemove: true,
+			PortBindings: {
+				'3306/tcp': [{ HostPort: `${port}` }],
+			},
+		},
+	});
+
+	await singlestoreContainer.start();
+	await new Promise((resolve) => setTimeout(resolve, 4000));
```

Comment:
I can foresee this maybe causing flake if the container takes long to start up. Are we fairly confident that 4 seconds is enough?

Reply:
@apeng-singlestore

Reply:
According to the official `dockerode` documentation, there is an `inspect` method that allow us to interact with the container, so we can implement a retry system, for example 5 retries in a span of 10 minutes, where we check if the container is already up and running, and if it's not, we sleep and check again in 2 minutes. 

Something like this: https://stackoverflow.com/questions/21183088/how-can-i-wait-for-a-docker-container-to-be-up-and-running 

This thread explains better the issue: https://github.com/apocas/dockerode/issues/191#issuecomment-156560628

Reply:
Solved by @apeng-singlestore